### PR TITLE
modified the IsLabelSelectorMatching function to reduce code duplication and fix typo

### PR DESCRIPTION
**Pull request by zehuaiWANG**
the solution for issue#4416~

### CODE REVIEW

**Comment by zehuaiWANG** [comment]
@floreks ，hi~ floreks， could you review this pr and the issue， thank you~

**Comment by floreks** [comment]
Rebase and fix Go tests.

File: src/app/backend/api/types.go [hunk 1]
```diff
@@ -219,33 +219,25 @@ var KindToAPIMapping = map[string]APIMapping{
...
-	}
-	for label, value := range selector {
-		if rsValue, ok := labelSelector.MatchLabels[label]; !ok || rsValue != value {
-			return false
-		}
-	}
-	return true
+// Resources(or subset) that a target object selector with the given selector.
+func IsLabelSelectorMatching(srcSelector map[string]string, targetLabelSelector *v1.LabelSelector) bool {
+  targetObjectLabels := targetLabelSelector.MatchLabels
```
Comment (floreks) [inline src/app/backend/api/types.go]: Add a check to see if `targetLabelSelector` pointer is not nil.

Comment (zehuaiWANG) [reply src/app/backend/api/types.go]: yes, i had added it now, could you have a look is it correct now?

Comment (zehuaiWANG) [reply src/app/backend/api/types.go]: @floreks thanks a lot~

**Comment by zehuaiWANG** [comment]
@floreks hi~floreks， I had added a check to see if targetLabelSelector pointer is not nil. Could you review it， thank you~

**Comment by zehuaiWANG** [comment]
@floreks

**Comment by floreks** [comment]
Tests fail. Not sure exactly why. Probably golinter fails.

**Comment by zehuaiWANG** [comment]
@floreks could you have a look if it's really now? I had used golint, and it seems that this is not the reason for failure, because he originally existed, thank you~
--------------------------------
types.go:144:1: exported method ResourceKind.Scalable should have comment or be unexported
types.go:180:6: exported type APIMapping should have comment or be unexported
types.go:180:6: type name will be used as api.APIMapping by other packages, and that stutters; consider calling this Mapping
types.go:189:1: comment on exported var KindToAPIMapping should be of the form "KindToAPIMapping ..."
---------------------------------------

**Comment by floreks** [comment]
There is an error in the logs.

```
[0] src/app/backend/api/types.go:241: File is not `goimports`-ed (goimports)
[0]   // If the resource has no selectors, then assume it targets different Pods.
[0]   if targetLabelSelector != nil {
[0]       targetObjectLabels := targetLabelSelector.MatchLabels
[0]       return IsSelectorMatching(srcSelector, targetObjectLabels)
[0]   }
[0]   return false
```

**Comment by zehuaiWANG** [comment]
@floreks yes, I see,  line 241 is a comment line, I had modified it. Could you have a look if it have anyother error ? Thank you~

**Comment by floreks** [comment]
You have access to the travis CI logs. Link is in the github checks. https://travis-ci.org/kubernetes/dashboard/jobs/598420568

Error is still the same as before.

**Comment by zehuaiWANG** [comment]
@floreks I looked at the log file. It seems that my line 241 has exited incorrectly, but my line 241 is a line of comments. I think it should not cause an error to exit. This looks strange, I deleted this line of comments to see if I can pass CI?

**Comment by zehuaiWANG** [comment]
@floreks could you have a look if the CI is ok？

**Comment by floreks** [comment]
This part is important:

`rc/app/backend/api/types.go:241: File is not `goimports`-ed (goimports)`

Format it

**Comment by zehuaiWANG** [comment]
@floreks hi~ I tried it many times and finally solved the problem caused by this format. Thank you for helping me.

File: src/app/backend/scaling/scale.go [hunk 2]
```diff
@@ -92,7 +92,7 @@ func getScaleGetter(cfg *rest.Config) (scale.ScalesGetter, error) {
 		return nil, err
 	}
 
-	cfg.GroupVersion = &apps.SchemeGroupVersion
+	cfg.GroupVersion = &appsv1beta2.SchemeGroupVersion
```
Comment (maciaszczykm) [inline src/app/backend/scaling/scale.go]: Why?

Comment (zehuaiWANG) [reply src/app/backend/scaling/scale.go]: I make a mistake, please close this pr, I will add a new one.

**Comment by floreks** [comment]
Why do you close, instead of simply rebasing/fixing existing one?

**Pull request closed without merging**